src/EditorFeatures/Core/Diagnostics/RawDiagnosticsTaggerProvider.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Shared.Preview;
+using Microsoft.CodeAnalysis.Editor.Shared.Tagging;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Editor.Tagging;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.CodeAnalysis.Workspaces;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+
+namespace Microsoft.CodeAnalysis.Diagnostics;
+
+/// <summary>
+/// Low level tagger responsible for producing specific diagnostics tags for some particular <see
+/// cref="RawDiagnosticTaggerConfiguration"/>.
+/// </summary>
+internal sealed class RawDiagnosticsTaggerProvider<TTag> : AsynchronousTaggerProvider<TTag>
+    where TTag : ITag
+{
+    private readonly RawDiagnosticTaggerConfiguration _configuration;
```

Comment:
this field is currently unused.  but we need to actually use this info to properly call into _analyzerService to compute the right set of diagnostics.